src/serac/physics/boundary_conditions/boundary_condition.hpp

```diff
@@ -36,18 +36,21 @@ class BoundaryCondition {
    * should be -1 for all components
    * @param[in] attrs The set of boundary condition attributes in the mesh that the BC applies to
    * @param[in] num_attrs The total number of boundary attributes for the mesh
+   * @param[in] state The finite element state on which this BC is applied
    */
   BoundaryCondition(GeneralCoefficient coef, const std::optional<int> component, const std::set<int>& attrs,
-                    const int num_attrs = 0);
+                    const int num_attrs = 0, FiniteElementState* state = nullptr);
 
   /**
    * @brief Minimal constructor for setting the true DOFs directly
    * @param[in] coef Either a mfem::Coefficient or mfem::VectorCoefficient representing the BC
    * @param[in] component The zero-indexed vector component if the BC applies to just one component,
    * should be -1 for all components
    * @param[in] true_dofs The indices of the relevant DOFs
+   * @param[in] state The finite element state on which this BC is applied
    */
-  BoundaryCondition(GeneralCoefficient coef, const std::optional<int> component, const mfem::Array<int>& true_dofs);
+  BoundaryCondition(GeneralCoefficient coef, const std::optional<int> component, const mfem::Array<int>& true_dofs,
+                    FiniteElementState* state = nullptr);
```

Comment:
I don't remember what our style guide says about raw pointers. Does `FiniteElementState* state` imply any changes in ownership?

Reply:
"Use raw pointers and references [only] to denote non-ownership (R.3 - R.4)"

https://serac.readthedocs.io/en/latest/sphinx/dev_guide/style_guide.html#code-style